packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1460,8 +1465,9 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
         newMode = FocusHighlightMode.traditional;
         break;
     }
-    if (newMode != _highlightMode) {
-      _highlightMode = newMode;
+    final FocusHighlightMode oldMode = highlightMode;
```

Comment:
Can the logic that determines to call notifyHighlightModeListeners() be in the setter for highlightMode?

Reply:
I don't want there to be a public setter for `highlightMode`, but I can make a private one, sure.

Reply:
Actually, adding a private setter overly complicates things, in my opinion, since it introduces yet another level of getters and setters (have to now have a __highlightMode or some such). Nothing else sets the private `_highlightMode` value, so it's not like there is duplication of logic.

I did add a comment to help explain it.